juriscraper/opinions/united_states/state/nh.py

```diff
@@ -1,127 +1,62 @@
-
-        (CASE 1)
-        The anchor has multiple lines broken with <br> tag(s), and each
-        line contains "<docket> <name>". In this case we need to combine
-        the docket numbers and name strings respectively.
-        [EXAMPLE: February 18, 2016 in nh_example_2.html]
-
-        (CASE 2)
-        The anchor has multiple lines broken with <br> tag(s), and the
-        second line is a continuation of a long case name started on the first
-        line.  So, the second line does not lead with a docket number, thus
-        this line's string should be glued onto the <name> substring extracted
-        from the previous line.
-        [EXAMPLE: September 18, 2018 in nh_example_6.html]
-        """
-        pairs = []
-        for anchor in self.html.xpath(self.link_path):
-            i = 0
-            dockets = []
-            name_substrings = []
-            text_anchor = anchor.text_content()
-            text_clean = text_anchor.replace("\n", "")
-
-            for text in text_clean.split(";"):
-                text = clean_string(text)
-                match = self.link_text_regex.search(text)
-                try:
-                    docket = match.group(1)
-                    dockets.append(docket)
-                    name = match.group(2)
-                    name = " ".join(name.split())
-                    name_substrings.append(name)
-                    i += 1
-                except AttributeError:
-                    if i == 0:
-                        # docket and name (root) should be contained in first substring
-                        error = f"Invalid anchor root string format: {text}"
-                        raise InsanityException(error)
-                    # no docket in the substring, its a trailing name substring
-                    # that they broke over multiple lines, so glue it to the
-                    # previous name substring
-                    name_substrings[i - 1] += f" {text}"
-            pairs.append(
+        self.url = "https://www.courts.nh.gov/content/api/documents"
+        self.parameters = {
+            "sort": "field_date_posted|desc",
+            "page": "1",
+            "size": "25",
+            # Opinions have a "documentPurposes" value of "1331|"
+            "purpose": "1331+undefined",
+            # Opinions have a "documentTags" value of "1206|"
+            "tag": "1206+",
+        }
+
+    def _download(self, request_dict={}):
+        if self.test_mode_enabled():
+            return super()._download(request_dict)
+        # Calls to API with POST are not allowed, using GET method
+        resp = self.request["session"].get(self.url, params=self.parameters)
+        return resp.json()
+
+    def _process_html(self) -> None:
+        for case in self.html["data"]:
+            docket, name = case["documentName"].split(",", 1)
+            date = case["documentPosted"]
+            url = re.findall(r"(http.*docu.*\.pdf)", case["documentContent"])[
```

Comment:
Can you do some more testing for the URL and docket, name generation.  

I was able to break the scraper by paging back only one page.   I think the split on , is too fragile.
Also - There is a case, `Hampstead School Board & a. v. School Administrative Unit No. 55` that you should look into.  

If you check out the url you can see its not an opinion its an order withdrawing the previous opinion from April.  The scraper assigns the new order the same values as the previous opinion.  And it extracts the url from the "related_document section" of the json response.  

This is close but it needs a little more testing and review.

Reply:
Added validation to only extract files for opinions.